src/ConnectionProps.ts

```diff
@@ -1,36 +1,36 @@
-export interface ConnectionProps {
-    /** The socket name. */
-    name?: string;
-    /** State IDs to always automatically subscribe to. */
-    autoSubscribes?: string[];
-    /** Automatically subscribe to logging. */
-    autoSubscribeLog?: boolean;
-    /** The protocol to use for the socket.io connection. */
-    protocol?: string;
-    /** The host name to use for the socket.io connection. */
-    host?: string;
-    /** The port to use for the socket.io connection. */
-    port?: string | number;
-    /** The socket.io connection timeout. */
-    ioTimeout?: number;
-    /** The socket.io command timeout. */
-    cmdTimeout?: number;
-    /** Flag to indicate if all objects should be loaded or not. Default true (not loaded) */
-    doNotLoadAllObjects?: boolean;
-    /** Flag to indicate if AccessControlList for current user will be loaded or not. Default true (not loaded) */
-    doNotLoadACL?: boolean;
-    /** Progress callback. */
-    onProgress?: (progress: number) => void;
-    /** Ready callback. */
-    onReady?: (objects: Record<string, ioBroker.Object>) => void;
-    /** Log callback. */
-    onLog?: (text: string) => void;
-    /** Error callback. */
-    onError?: (error: any) => void;
-    /** Object change callback. */
-    onObjectChange?: ioBroker.ObjectChangeHandler;
-    /** Language callback */
-    onLanguage?: (lang: ioBroker.Languages) => void;
-    /** Forces the use of the Compact Methods, wich only exists in admin 5 UI. */
-    admin5only?: boolean;
-}
+export interface ConnectionProps {
```

Comment:
Why is this called "...Props" and not "...Options" as we do for everything else?

Reply:
This library was extracted from `adapter-react`, so I guess that's why.
In this first version of the rewrite I didn't want to do any breaking changes (except for the one where two identical methods on the two classes behaved differently).

I agree that it should be called `ConnectionOptions` and I think some of the params need to be renamed. But that's for a later revision: https://github.com/AlCalzone/socket-client/issues/1